src/collections/cursor.ts

```diff
@@ -35,7 +35,7 @@ export class FindCursor {
         this.options = options ?? {};
 
         const isOverPageSizeLimit = this.options.sort &&
-            this.options.sort.$vector == null &&
+            !('$vector' in this.options.sort) &&
```

Comment:
I believe we need to check if $vectorize is not present as well. `!('$vectorize' in this.options.sort) &&`?